### ISSUE TITLE
Related managers not filtering on deleted field

### ISSUE THREAD

**Issue by fatalispm** [makinacorpus/django-safedelete#166]
I am not sure if this is the desired behavior:

```
class User(models.Model): 
     ...
     
class PropertyQS(SafeDeleteQueryset, SomeOtherQueryset):
     ....
class Property(SafeDeleteModel):
       objects = SafeDeleteManager.from_queryset(...)
       owner = models.ForeignKey(User)
```

Now if I do:
    
```
    user.own_properties.all() it returns a simple query which looks like this:
    
    SELECT * FROM "property" WHERE "property"."owner_id" = 2 
    also returning deleted properties. 
```
    
It works the same way even if the User model is safedeletable too. 
Is there a way to enforce filtering deleted objects on related models too?
       
 This causes problems with SafeDeletable M2M fields, since user. properties.remove(property) will just do nothing.

**Comment by Gagaro** [comment]
Can you test with the master branch? This may have been solved with https://github.com/makinacorpus/django-safedelete/pull/160

**Comment by fatalispm** [comment]
@Gagaro thanks, it works on master. 

Btw, it might break some of the functionality if you are using multiple QuerySets, for example, CTEQuerySet, since they both override the query object. 
Solved it the following way:

`
class SafeDeleteCTEQuery(SafeDeleteQuery, CTEQuery):
    pass 
`

And then in qs you can init it with this query.

**Comment by Gagaro** [comment]
Yes, the same way you would need to have your own manager if you use several custom ones.

I released the fix in 1.1.0.

**Comment by fatalispm** [comment]
@Gagaro thanks! 

It is not exactly the same though, since you have to override your queryset's __init__ method with this query, which was not required in the previous releases since back then you didn't override the query object of the queryset. Otherwise, your code will fail. I thought it is worth mentioning for future readers.
